api/src/main/java/org/apache/iceberg/transforms/Timestamps.java

```diff
@@ -54,12 +55,17 @@ public Integer apply(Long timestampMicros) {
 
     // discards fractional seconds, not needed for calculation
     OffsetDateTime timestamp = Instant
-        .ofEpochSecond(timestampMicros / 1_000_000)
+        .ofEpochSecond(timestampMicros / 1_000_000 + zoneOffset.getTotalSeconds())
         .atOffset(ZoneOffset.UTC);
 
     return (int) granularity.between(EPOCH, timestamp);
   }
 
+  public Timestamps zoneOffset(ZoneOffset newZoneOffset) {
+    this.zoneOffset = newZoneOffset;
```

Comment:
This changes the zone offset for all uses of the given granularity in this JVM and can unexpectedly change the behavior of another table. I think it would be a serious correctness bug.

Instead, I think this PR needs to create a new `TimestampTransform` class like the `BucketTransform` class and remove old references to this enum.

Reply:
>This changes the zone offset for all uses of the given granularity in this JVM and can unexpectedly change the behavior of another table. I think it would be a serious correctness bug.

The variable zoneOffset is not a global static variable,I'm not sure why it affects the other tables. 
But I may have realized that enumerating classes is not really a good way to use them.